server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -198,6 +198,21 @@ public Response createOrUpdateSecret(@Auth AutomationClient automationClient,
     return Response.created(uriBuilder.build()).build();
   }
 
+  @Timed @ExceptionMetered
+  @Path("{oldName}/rename/{newName}")
+  @POST
+  @Consumes(APPLICATION_JSON)
+  public Response renameSecret(
+      @Auth AutomationClient automationClient,
+      @PathParam("oldName") String oldName,
+      @PathParam("newName") String newName) {
```

Comment:
I'm not terribly familiar with @Consumes, but the precedent seems to be that even if the POST method takes no input other than URL parameters it still declares that it consumes JSON. On the client side, the request type is still application/json (or whatever) but the body is empty.